### PR TITLE
Add logic xor and arithmetic ^

### LINKED ISSUE #636: nGQL expressions support 'or' 'and' 'not'
Just like mysql, nGQL logic operators support _"And"_, _"Or"_ and _"Not"_ etc.

### PR DESCRIPTION
**Pull request by laura-ding**
close #636

### CODE REVIEW

**Comment by laura-ding** [comment]
jenkins go

File: src/common/filter/Expressions.cpp [hunk 1]
```diff
@@ -839,10 +842,22 @@ OptVariantType ArithmeticExpression::eval() const {
...
+            if (isArithmetic(l) && isArithmetic(r)) {
+                if (isDouble(l) || isDouble(r)) {
+                    return fmod(asDouble(l), asDouble(r));
+                }
                 return OptVariantType(asInt(l) % asInt(r));
             }
             break;
+        case XOR:
+            if (isArithmetic(l) && isArithmetic(r)) {
+                if (isDouble(l) || isDouble(r)) {
```
Comment (critical27) [inline src/common/filter/Expressions.cpp]: Do we really need to support xor for double? It is a little bit weird that we round up double and then do xor.

Comment (laura-ding) [reply src/common/filter/Expressions.cpp]: I see mysql is supported

Comment (darionyaphet) [reply src/common/filter/Expressions.cpp]: What scenarios are applicable for xor with double ?

**Comment by darionyaphet** [comment]
LGTM ! Could you support some document to explain XOR operator ?

File: src/common/filter/Expressions.cpp (hunk 1 above)
Comment (dutor) [reply src/common/filter/Expressions.cpp]: @critical27 @darionyaphet 
Personally, I have not seen this usage.

But from the perspective of semantic of XOR itself, or other bit ops like bitwise AND, they are just literally performing the ops bitwisely. It's up to the user to the determine whether the op is meaningful.

**Pull request merged**